azurerm/resource_arm_key_vault_test.go

```diff
@@ -648,3 +672,69 @@ resource "azurerm_key_vault" "test" {
 }
 `, rInt, location, rInt)
 }
+
+func testAccAzureRMKeyVault_accessPolicyUpperLimit(rInt int, location string, rs string) string {
+
+	var storageAccountConfigs string
+	var accessPoliciesConfigs string
+
+	for i := 1; i <= 20; i++ {
+		storageAccountConfigs += testAccAzureRMKeyVault_generateStorageAccountConfigs(i, rs)
+		accessPoliciesConfigs += testAccAzureRMKeyVault_generateAccessPolicyConfigs(i)
+	}
+
+	return fmt.Sprintf(`
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_key_vault" "test" {
+  name                = "vault%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  tenant_id           = "${data.azurerm_client_config.current.tenant_id}"
+
+  sku {
+    name = "premium"
+  }
+%s
+}
+
+%s
+`, rInt, location, rInt, accessPoliciesConfigs, storageAccountConfigs)
+}
+
+func testAccAzureRMKeyVault_generateStorageAccountConfigs(accountNum int, rs string) string {
+	return fmt.Sprintf(`
+resource "azurerm_storage_account" "testsa%d" {
+    name                      = "testsa%d%s"
```

Comment:
_minor_ could we swap these values:
```suggestion
    name                      = "testsa%s%d"
```